src/libs/ajax/AzureStorage.ts

```diff
@@ -111,6 +111,51 @@ export const AzureStorage = (signal?: AbortSignal) => ({
     }), notebooks)
   },
 
+  blobMetadata: (azureStorageUrl: string) => {
+    const getObjectMetadata = async () => {
+      const workspaceId = azureStorageUrl.split('/')[3].replace('sc-', '')
+      const fileName = _.last(azureStorageUrl?.split('/'))?.split('.')?.join('.')
+      // check if file can just be fetched without sas token
```

Comment:
I would assume that the vast majority of files in Terra will not be public. Does it make sense to try this first for every file?

Reply:
that is likely, however, I am having difficulty thinking through how to properly display errors in the reverse flow. this flow felt more natural since the parent class handles it for me, if I revert the order I'd need to save the first error in case the file is not public and the error message it  gives for public vs not is different -- or would it be okay to assume that if both fail, it would be reasonable to defer to the second failure?

Reply:
switched the logic around.